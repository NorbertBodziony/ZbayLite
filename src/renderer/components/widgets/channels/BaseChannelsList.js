import React from 'react'
import Immutable from 'immutable'
import PropTypes from 'prop-types'

import List from '@material-ui/core/List'

import ChannelsListItem from '../../../containers/widgets/channels/ChannelsListItem'
import OfferListItem from '../../../containers/widgets/channels/OfferListItem'

export const propTypes = {
  channel: PropTypes.shape({
    name: PropTypes.string.isRequired,
    private: PropTypes.bool.isRequired,
    hash: PropTypes.string.isRequired,
    address: PropTypes.string.isRequired,
    unread: PropTypes.number,
    description: PropTypes.string
  })
}

export const BaseChannelsList = ({
  channels,
  directMessages,
  selected,
  offers,
  selectedOffer
}) => {
  const [...keys] = offers.keys()
  return (
    <List disablePadding>
      {channels.map(channel => (
        <ChannelsListItem
          key={directMessages ? channel.get('address') : channel.get('id')}
          channel={channel}
          directMessages={directMessages}
          selected={selected}
        />
      ))}
      {offers.toList().map((offer, index) => (
        <OfferListItem
          key={keys[index]}
          channel={offer}
          selected={selectedOffer}
        />
      ))}
    </List>
  )
}

BaseChannelsList.propTypes = {
  channels: PropTypes.instanceOf(Immutable.List).isRequired,
  selected: PropTypes.instanceOf(Immutable.Record).isRequired,
  selectedOffer: PropTypes.instanceOf(Immutable.Record).isRequired,
  directMessages: PropTypes.bool
}

BaseChannelsList.defaultProps = {
  channels: Immutable.List(),
  offers: Immutable.Map(),
  displayAddress: false,
  directMessages: false
}

export default React.memo(BaseChannelsList)
