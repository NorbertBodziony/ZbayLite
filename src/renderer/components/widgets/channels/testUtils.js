export const createChannel = id => ({
  name: `Channel ${id}`,
  description: id % 2 === 0 ? '' : `Channel about ${id}`,
  private: id % 2 === 0,
  unread: id,
  hash: `test-hash-${id}`,
  address: `zs1testaddress${id}`
})
