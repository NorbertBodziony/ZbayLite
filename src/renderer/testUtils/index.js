import { DateTime, Settings } from 'luxon'
import BigNumber from 'bignumber.js'

import { messages as zbayMessages } from '../zbay'

Settings.defaultZoneName = 'utc'

export const now = DateTime.utc(2019, 3, 7, 13, 3, 48)

export const createChannel = id => ({
  id: id,
  name: `Channel ${id}`,
  private: Boolean(id % 2),
  address: `zs1testaddress${id}`,
  unread: id,
  description: id % 2 === 0 ? '' : `Channel about ${id}`,
  keys: {
    ivk: `incoming-viewing-key-${id}`
  }
})

export const channels = {
  createChannel
}

export const createMessage = (
  id,
  createdAt = now.minus({ hours: id }).toSeconds(),
  replyTo = 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya'
) => ({
  id,
  type: zbayMessages.messageType.BASIC,
  createdAt,
  message: `This is some message ${id}`,
  sender: {
    replyTo
  }
})

export const createReceivedMessage = ({
  id,
  createdAt = now.toSeconds()
}) => ({
  id,
  spent: new BigNumber('0.32'),
  type: zbayMessages.messageType.BASIC,
  createdAt,
  message: `This is a message with id ${id}`,
  sender: {
    replyTo: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  }
})

export const createSendableMessage = ({
  message,
  createdAt = now.toSeconds()
}) => ({
  type: zbayMessages.messageType.BASIC,
  sender: {
    replyTo: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  createdAt,
  message
})

export const createSendableTransferMessage = ({
  message,
  createdAt = now.toSeconds()
}) => ({
  type: zbayMessages.messageType.BASIC,
  sender: {
    replyTo: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  receiver: {
    replyTo: 'zs1z9rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Sun'
  },
  createdAt,
  message
})

export const createReceivedTransferMessage = ({ id, createdAt = now.toSeconds() }) => ({
  id,
  spent: new BigNumber('0.32'),
  type: zbayMessages.messageType.TRANSFER,
  createdAt,
  message: `This is a message with id ${id}`,
  sender: {
    replyTo: 'zs1z7rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Wenus'
  },
  receiver: {
    replyTo: 'zs1z9rejlpsa98s2rrrfkwmaxu53e4ue0ulcrw0h4x5g8jl04tak0d3mm47vdtahatqrlkngh9slya',
    username: 'Sun'
  }
})

export const messages = {
  createReceivedMessage,
  createSendableMessage,
  createReceivedTransferMessage,
  createSendableTransferMessage,
  createMessage
}

export const createTransfer = ({
  txid,
  amount = '0.0001',
  memo = '',
  change = false
}) => ({
  txid,
  amount,
  memo,
  change
})

export const transfers = {
  createTransfer
}

export const createIdentity = ({
  name = 'Saturn',
  address = 'saturn-private-address',
  transparentAddress = 'saturn-transparent-address',
  keys = {
    tpk: 'saturn-tpk',
    sk: 'saturn-sk'
  }
} = {}) => ({
  name,
  address,
  transparentAddress,
  keys
})

export default {
  now,
  transfers,
  channels,
  messages,
  createIdentity
}
