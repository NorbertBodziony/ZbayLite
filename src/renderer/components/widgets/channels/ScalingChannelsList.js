import React from 'react'
import Immutable from 'immutable'
import PropTypes from 'prop-types'

import BaseChannelsList from './BaseChannelsList'

export const ScalingChannelsList = ({ channels, maxHeight, displayAddress, selected }) => {
  return <BaseChannelsList height={maxHeight} channels={channels} displayAddress={displayAddress} selected={selected} />
}

ScalingChannelsList.propTypes = {
  maxHeight: PropTypes.number.isRequired,
  channels: PropTypes.instanceOf(Immutable.List).isRequired,
  selected: PropTypes.instanceOf(Immutable.Record).isRequired,
  displayAddress: PropTypes.bool
}

ScalingChannelsList.defaultProps = {
  channels: Immutable.List()
}

export default React.memo(ScalingChannelsList)
