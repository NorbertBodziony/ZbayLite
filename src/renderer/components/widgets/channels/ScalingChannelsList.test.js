import React from 'react'
import Immutable from 'immutable'
import { shallow } from 'enzyme'
import * as R from 'ramda'

import { ScalingChannelsList } from './ScalingChannelsList'
import { constants } from './BaseChannelsList'
import testUtils from '../../../testUtils'

describe('ScallingChannelsList', () => {
  it('renders component', () => {
    const channels = Immutable.fromJS(
      R.range(0, 4).map(testUtils.channels.createChannel)
    )
    const targetHeight = constants.itemSize * 4
    const result = shallow(
      <ScalingChannelsList channels={channels} maxHeight={targetHeight + 20} />
    )
    expect(result.prop('height')).toEqual(targetHeight)
    expect(result).toMatchSnapshot()
  })
})
