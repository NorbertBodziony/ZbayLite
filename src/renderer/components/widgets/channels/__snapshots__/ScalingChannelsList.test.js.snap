// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScallingChannelsList renders component 1`] = `
<Memo(BaseChannelsList)
  channels={
    Immutable.List [
      Immutable.Map {
        "id": 0,
        "name": "Channel 0",
        "private": false,
        "hash": "test-hash-0",
        "address": "zs1testaddress0",
        "unread": 0,
        "description": "",
        "keys": Immutable.Map {
          "ivk": "incoming-viewing-key-0",
        },
      },
      Immutable.Map {
        "id": 1,
        "name": "Channel 1",
        "private": true,
        "hash": "test-hash-1",
        "address": "zs1testaddress1",
        "unread": 1,
        "description": "Channel about 1",
        "keys": Immutable.Map {
          "ivk": "incoming-viewing-key-1",
        },
      },
      Immutable.Map {
        "id": 2,
        "name": "Channel 2",
        "private": false,
        "hash": "test-hash-2",
        "address": "zs1testaddress2",
        "unread": 2,
        "description": "",
        "keys": Immutable.Map {
          "ivk": "incoming-viewing-key-2",
        },
      },
      Immutable.Map {
        "id": 3,
        "name": "Channel 3",
        "private": true,
        "hash": "test-hash-3",
        "address": "zs1testaddress3",
        "unread": 3,
        "description": "Channel about 3",
        "keys": Immutable.Map {
          "ivk": "incoming-viewing-key-3",
        },
      },
    ]
  }
  height={148}
/>
`;
